### ISSUE TITLE
[naming-conventions] False negative with variableLike.

### ISSUE THREAD

**Issue by eestein** [typescript-eslint/typescript-eslint#2145]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/naming-convention": [
            "error",
            {
                "selector": "variableLike",
                "format": [
                    "camelCase"
                ]
            },
            {
                "selector": "memberLike",
                "modifiers": [
                    "private",
                    "protected"
                ],
                "format": [
                    "camelCase"
                ]
            },
            {
                "selector": "memberLike",
                "modifiers": [
                    "public",
                    "static",
                    "readonly",
                    "abstract"
                ],
                "format": [
                    "camelCase",
                    "PascalCase"
                ]
            },
            {
                "selector": "enumMember",
                "format": [
                    "PascalCase"
                ]
            },
            {
                "selector": "variable",
                "types": [
                    "boolean"
                ],
                "format": [
                    "PascalCase"
                ],
                "prefix": [
                    "is",
                    "are",
                    "should",
                    "has",
                    "can",
                    "did",
                    "will"
                ]
            },
            {
                "selector": "interface",
                "format": [
                    "PascalCase"
                ],
                "custom": {
                    "regex": "^I[A-Z]",
                    "match": false
                }
            },
            {
                "selector": "typeLike",
                "format": [
                    "PascalCase"
                ]
            }
        ],
  }
}
```

```TS
import { Component } from '@angular/core';

@Component({
    selector: 'app-root',
    templateUrl: './app.component.html',
    styleUrls: ['./app.component.scss']
})
export class AppComponent {
    title = 'aloha';
    public ALAMBRA = 3;
    private testing = ';';
    private AlhaBrand = '';
    private ALOHA = 3;

    private SETL = () => '';

    private TEST(ALOHA: any): any {
        const PascalCase = ALOHA;
        return PascalCase;
    }
}

function TESTING() {

}
```

**Expected Result**
Errors on `ALAMBRA`, `AlhaBrand`, `ALOHA` (property), `ALOHA` (parameter), `PascalCase`(variable), `SETL`, `TEST` and `TESTING` functions.

**Actual Result**
Errors only on `ALOHA` (parameter), `PascalCase`(variable)  and `TESTING` function.

Please see image:
![Screen Shot 2020-06-01 at 09 46 22](https://user-images.githubusercontent.com/2049255/83411090-f5860880-a3ed-11ea-9c67-79f13f1b9cc3.jpg)

**Additional info**
If I use `default` instead of `variableLike` it throws the error. But that's not what I want. I'd like to use specific rules for TS functions/properties.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.0.2` |
| `@typescript-eslint/parser`        | `3.0.2` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `14.3.0` |
| `npm`                              | `6.14.5` |

Is it the rule or did I set it up incorrectly? I tried to follow the docs to the letter and did not see any specific rules for TS functions/properties.

**Comment by bradzacher** [comment]
You've got your config incorrect.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#selector-options

> `modifiers` allows you to specify which modifiers to granularly apply to
> - The name must match all of the modifiers.

Your config specifies things like this
```
            {
                "selector": "memberLike",
                "modifiers": [
                    "private",
                    "protected"
                ],
                "format": [
                    "camelCase"
                ]
            },
```
Which is impossible to match - you can never have a private protected member.
If you want to enforce naming conventions for private or protected, they must be two separate configs.

**Comment by eestein** [comment]
Thank you, I missed that important detail.
